src/workerd/api/memory-cache.c++

```diff
@@ -239,14 +239,26 @@ SharedMemoryCache::Use::~Use() noexcept(false) {
 }
 
 kj::Maybe<kj::Own<CacheValue>> SharedMemoryCache::Use::getWithoutFallback(
-    const kj::String& key) const {
-  auto data = cache->data.lockExclusive();
+    const kj::String& key, SpanBuilder& span) const {
+  kj::Locked<ThreadUnsafeData> data;
+  {
```

Comment:
I am not super fond of using lambdas for variable initialization, but this case will benefit from it imo:

```
data = [&span]() { ... }();
```